src/components/ShoppingLists/ShoppingLists.js

```diff
@@ -0,0 +1,174 @@
+
+const target = GRAPHQL_TARGETS.COMMERCETOOLS_PLATFORM;
+
+const ShoppingLists = () => {
+  const [shoppingLists, setShoppingLists] = useState([{}]);
+  const [formVisibility, setFormVisibility] = useState(false);
+  const showApiErrorNotification = useShowApiErrorNotification();
+  const showSuccessNotification = useShowNotification();
+  const [saveShoppingList] = useMutation(createShoppingList);
+  const [delShoppingList] = useMutation(deleteShoppingList);
+  const cols = [
+    {
+      key: "id",
+      label: "id",
+      renderItem: (row) => (row?.id ? row.id : ""),
+    },
+    {
+      key: "version",
+      label: "version",
+      renderItem: (row) => (row?.version ? row.version : ""),
+    },
+    {
+      key: "name",
+      label: "Name",
+      renderItem: (row) =>
+        row?.nameAllLocales ? row?.nameAllLocales[0]?.value : "",
+    },
+    {
+      key: "actions",
+      label: "Actions",
+      renderItem: (row) => {
+        const version = row.version;
+        const id = row.id;
+        return (
+          <IconButton
+            icon={<CheckInactiveIcon />}
+            label="Delete Record"
+            onClick={async (row) => {
+              const { error, data } = await delShoppingList({
+                variables: {
+                  version,
+                  id,
+                },
+                context: { target },
+              });
+              if (error) showApiErrorNotification({ error: error });
+              console.log(data);
+              setShoppingLists(
+                shoppingLists.filter((item) => {
+                  console.log(data);
+                  return item.id !== data.deleteShoppingList.id;
+                })
+              );
+              showSuccessNotification({
+                kind: "success",
+                domain: DOMAINS.SIDE,
+                text: "Shopping List Deleted Successfully",
+              });
+            }}
+          />
+        );
+      },
+    },
+  ];
+  const formik = useFormik({
+    initialValues: {
+      locale: "",
+      name: "",
+    },
+    onSubmit: (values, { resetForm }) => {
+      handleSubmit(values);
+      resetForm({});
+    },
+    validateOnChange: false,
+    validateOnBlur: false,
+  });
+
+  const { error, data, loading } = useQuery(fetchShoppingLists, {
+    context: { target },
+  });
+
+  const handleSubmit = async (formValues) => {
+    const { error, data, loading } = await saveShoppingList({
+      variables: {
+        locale: formValues.locale,
+        value: formValues.name,
+      },
+      context: { target },
+    });
+    if (error) showApiErrorNotification({ error: error });
+    setShoppingLists([...shoppingLists, data.createShoppingList]);
+    setFormVisibility(false);
+    showSuccessNotification({
+      kind: "success",
+      domain: DOMAINS.SIDE,
+      text: "Shopping List added Successfully",
+    });
+  };
+
+  useEffect(() => {
+    setShoppingLists(data?.shoppingLists?.results);
+  }, [data]);
```

Comment:
This does not look right. There is also a risk of ending up in an infinite loop.

Remove it, as you also don't need a state for this.